src/shared_misc.hpp

```diff
@@ -26,7 +26,10 @@
 #include <locale>
 #include <type_traits>
 #include <forward_list>
-#include <sys/sysctl.h>
+
+#if defined(__APPLE__)
```

Comment:
```suggestion
#if __has_include(<sys/sysctl.h>)
```

This header is also used by FreeBSD and potentially other systems; just not my Linux any more. Can you check if the proposed change works? It might still pick up the header on Linux, but it shouldn't...

Reply:
works